src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/quantified-associations/components/QuantifiedAssociationRow.tsx

```diff
@@ -112,7 +112,10 @@ const QuantifiedAssociationRow = ({
   const translate = useTranslate();
   const {isGranted} = useSecurity();
   const isProductModel = ProductType.ProductModel === row.productType;
-  const productEditUrl = useRoute(`pim_enrich_${row.productType}_edit`, {id: row.product?.id.toString() || ''});
+  const routeParams = isProductModel
+    ? {id: row.product?.id.toString() || ''}
+    : {uuid: row.product?.id.toString() || ''};
+  const productEditUrl = useRoute(`pim_enrich_${row.productType}_edit`, routeParams);
```

Comment:
In order to fix your CI problem you can change thoses lines by: 
```
- import {filterErrors, formatParameters, useTranslate, useSecurity, useRoute} from '@akeneo-pim-community/shared';
+ import {filterErrors, formatParameters, useTranslate, useSecurity, useRouter} from '@akeneo-pim-community/shared';
```
and
```suggestion
  const router = useRouter();
  const productEditUrl = isProductModel
    ? router.generate('pim_enrich_product_model_edit', {id: row.product?.id.toString() || ''})
    : router.generate('pim_enrich_product_edit', {uuid: row.product?.id.toString() || ''});
```